tests/unit/test_operator.py

```diff
@@ -154,3 +244,79 @@ def test_pebble_services_running(
     )
     assert environment["BACKEND_MODE"] == harness.charm.config.get("backend-mode")
     assert environment["VOLUME_VIEWER_IMAGE"] == harness.charm.config.get("volume-viewer-image")
+
+
+def test_istio_relations_conflict_detector_no_relations(
```

Comment:
suggestion (non-blocking): these 4 tests from this point onwards share most of the logic, can we parametrize them into a unique one with booleans for enabled relationships - defining test cases - and conditional checks for acting and asserting?